tests/Elasticsearch/Tests/ClientTest.php

```diff
@@ -88,136 +94,148 @@ public function testFromConfigBadParamQuiet()
+        $this->expectExceptionMessage('type cannot be an empty string');
 
-        try {
-            $client->delete([
-                'index' => 'test',
-                'type' => ['', '', ''],
-                'id' => 'test'
-            ]);
-            $this->fail("InvalidArgumentException was not thrown");
-        } catch (Elasticsearch\Common\Exceptions\InvalidArgumentException $e) {
-            // all good
-        }
+        $client->delete([
+            'index' => 'test',
+            'type' => '',
+            'id' => 'test'
+        ]);
     }
 
-    public function testArrayOfNullDelete()
+    public function testIdCannotBeEmptyStringForDelete()
     {
         $client = ClientBuilder::create()->build();
 
-        try {
-            $client->delete([
-                'index' => [null, null, null],
-                'type' => 'test',
-                'id' => 'test'
-            ]);
-            $this->fail("InvalidArgumentException was not thrown");
-        } catch (Elasticsearch\Common\Exceptions\InvalidArgumentException $e) {
-            // all good
-        }
+        $this->expectException(Elasticsearch\Common\Exceptions\InvalidArgumentException::class);
+        $this->expectExceptionMessage('id cannot be an empty string');
 
-        try {
-            $client->delete([
-                'index' => 'test',
-                'type' => [null, null, null],
-                'id' => 'test'
-            ]);
-            $this->fail("InvalidArgumentException was not thrown");
-        } catch (Elasticsearch\Common\Exceptions\InvalidArgumentException $e) {
-            // all good
-        }
+        $client->delete([
+            'index' => 'test',
+            'type' => 'test',
+            'id' => ''
+        ]);
+    }
+
+    public function testIndexCannotBeArrayOfEmptyStringsForDelete()
+    {
+        $client = ClientBuilder::create()->build();
+
+        $this->expectException(Elasticsearch\Common\Exceptions\InvalidArgumentException::class);
+        $this->expectExceptionMessage('index cannot be an array of empty strings');
+
+        $client->delete([
+            'index' => ['', '', ''],
+            'type' => 'test',
+            'id' => 'test'
+        ]);
+    }
+
+    public function testTypeCannotBeArrayOfEmptyStringsForDelete()
+    {
+        $client = ClientBuilder::create()->build();
+
+        $this->expectException(Elasticsearch\Common\Exceptions\InvalidArgumentException::class);
+        $this->expectExceptionMessage('type cannot be an array of empty strings');
+
+        $client->delete([
+            'index' => 'test',
+            'type' => ['', '', ''],
+            'id' => 'test'
+        ]);
+    }
+
+    public function testIndexCannotBeArrayOfNullsForDelete()
+    {
+        $client = ClientBuilder::create()->build();
+
+        $this->expectException(Elasticsearch\Common\Exceptions\InvalidArgumentException::class);
+        $this->expectExceptionMessage('index cannot be an array of empty strings');
```

Comment:
The message is slightly incorrect, but I don't think it is necessary to change it. (the code does not differ between null and empty string https://github.com/mhujer/elasticsearch-php/blob/132e291fb2e36dad5c6c538ecb6e1cbfa16d2fd8/src/Elasticsearch/Client.php#L1553)